### PR TITLE
feat: support cross-attn cuda graph for whisper

### PR DESCRIPTION
**Pull request by mickqian**
Previously, whisper model use custom `bmm + mask` implementation of cross-attn, which make it in-compatible with cuda graph

## Modifications

 - Reworked cross-attention to use native `RadixAttention` path instead of the custom `bmm + mask` implementation.
  - Removed the Python-side Whisper encoder cache, switched encoder outputs to the native encoder-decoder cache flow managed by scheduler/attention backends.
  - Fixed CUDA graph replay padding for encoder-decoder batches by initializing replay buffers with valid encoder length metadata.
  - Narrowed CUDA graph enablement in `server_args`: explicitly allow Whisper only when both prefill and decode use`flashinfer`
  - Fixed `flashinfer` decode cross-attention CUDA-graph planning to use encoder-side CPU lengths, instead of incorrectly reusing decoder `seq_lens_cpu`

## Accuracy Tests

```
# launch server
$ sglang serve \
    --model-path openai/whisper-tiny \
    --attention-backend flashinfer \
    --host 0.0.0.0 \
    --port 31018 \
    --cuda-graph-max-bs 2

# benchmark
$ python benchmark/asr/bench_sglang.py \
    --base-url http://127.0.0.1:31011 \
    --model openai/whisper-tiny \
    --api-type transcription \
    --language en \
    --concurrency 1 \
    -n 100
```

  - no-cg: WER 176.8765, avg latency 0.0980s, throughput 8.96 req/s
  - cuda-graph: WER 176.8765, avg latency 0.0503s, throughput 15.60 req/s
  
  
  

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/models/whisper.py [hunk 1]
```diff
@@ -99,65 +99,16 @@ def forward(
...
-                                q_start = q_end
-                            attn_weights = attn_weights.masked_fill(
-                                ~mask.unsqueeze(0), float("-inf")
-                            )
-
-            attn_weights = torch.nn.functional.softmax(attn_weights, dim=-1)
-            attn_output = torch.bmm(attn_weights, v)
-            attn_output = attn_output.transpose(0, 1)
-            attn_output = attn_output.reshape(q_len, num_heads * head_dim)
+            q = q.view(-1, self.attn.tp_q_head_num, self.attn.head_dim)
```
Comment (mickqian) [inline python/sglang/srt/models/whisper.py:106]: replaced hand-written cross-attn with native backend (RadixAttention)

File: python/sglang/srt/layers/attention/flashinfer_backend.py [hunk 2]
```diff
@@ -1053,11 +1053,17 @@ def update_cross_attention(
                 # Normal attention
                 paged_kernel_lens = seq_lens
                 kv_start_idx = encoder_lens
+                paged_kernel_lens_cpu = seq_lens_cpu
             else:
                 # Cross attention
                 paged_kernel_lens = encoder_lens
                 kv_start_idx = torch.zeros_like(encoder_lens)
                 seq_lens_sum = encoder_lens.sum().item()
+                paged_kernel_lens_cpu = (
```
Comment (mickqian) [inline python/sglang/srt/layers/attention/flashinfer_backend.py:1062]: should be the encoder length in encoder-decoder arch
